src/net-ipfs-engine/Catalyst.Ipfs.Http.Server/HttpApi/V0/IpfsController.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.IO;
+using System.Text;
+using System.Threading;
+using Catalyst.Ipfs.Core.CoreApi;
+using Lib.P2P;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
+
+namespace Catalyst.Ipfs.Http.Server.HttpApi.V0
+{
+    /// <summary>
+    ///   A base controller for IPFS HTTP API.
+    /// </summary>
+    /// <remarks>
+    ///   Any unhandled exceptions are translated into an <see cref="ApiError"/> by the
+    ///   <see cref="ApiExceptionFilter"/>.
+    /// </remarks>
+    [Route("api/v0")]
+    [Produces("application/json")]
+    [ApiExceptionFilter]
+    public abstract class IpfsController : Controller
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Convert this 'abstract' class to a concrete class with a protected constructor.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)